src/DependencyInjection/SonataClassificationExtension.php

```diff
@@ -107,131 +111,145 @@ public function registerDoctrineMapping(array $config)
 
         $collector = DoctrineCollector::getInstance();
 
-        $collector->addAssociation($config['class']['category'], 'mapOneToMany', [
-            'fieldName' => 'children',
-            'targetEntity' => $config['class']['category'],
-            'cascade' => [
+        $categoryOnyToManyOptions = OptionsBuilder::create()
+            ->add('fieldName', 'children')
```

Comment:
I wouldn't use this signature to create a mapping, because you must know the technical details.

There is an issue with a better naming approach: https://github.com/sonata-project/sonata-doctrine-extensions/issues/172